datafusion/common/src/config.rs

```diff
@@ -250,6 +250,11 @@ config_namespace! {
         /// specified. The Arrow type system does not have a notion of maximum
         /// string length and thus DataFusion can not enforce such limits.
         pub support_varchar_with_length: bool, default = true
+
+        /// When set to true, the source locations relative to the original SQL
```

Comment:
Is the idea that we will keep this off by default at first as we evaluate the impact it has on performance?

Reply:
Yeah and also until we produce diagnostics for more of the most common errors, or so I thought. Right now it's very few errors.

Reply:
And that's okay I think, I wanted this PR to be more digestible and a PoC for what can be achieved with `Diagnostic`. Rolling it out to more places will come later.